useImageColour Hook - node-vibrant (#9683)
bbc/simorgh: src/app/hooks/useImageColour/README.md

- This hook has some fallback behaviour to more gracefully deal with error states

## Client Side Only

This hook currently only runs on the client side.

This does mean the colour will not be available on first page load - instead, the colour will initially use a customisable fallback colour. The hook also returns a `loading: true` boolean in this scenario, for more options on how to deal with the delay.

## Basic Usage

```javascript
import useImageColour from './';

const MyComponent = () => {
  const output = useImageColour('http://placekitten.com/200/300');

  return JSON.stringify(output);
};
```

## Input

The first argument to the hook is a link to the image to be used.

- For us to be able to download the image, the server must respond with appropriate CORS headers. BBC services, including iChef, do include the required headers.
- Any image size beyond approximately 20 x 20 pixels is unnecessary. Unless a larger version of the image is being downloaded anyway for some other use, generally using a very small image will conserve bandwidth

##### `options`

The hook accepts a second, optional argument. This argument can contain various configuration options:

> - `contrastColour` - A hexidecimal colour string, eg `"#ff0005"` or RGB array, eg `[255, 0, 5]`. If provided, each colour in the `colours` array returned from the hook will include a `contrast` number that compares the contrast of the generated colour to this `contrastColour`. This calculation uses the process described here: https://www.w3.org/TR/WCAG20-TECHS/G17.html

Author: ```suggestion
- `contrastColour` - A hexidecimal colour string, eg `"#ff0005"`. If provided, this colour will be used in the assessment of contrast requirements, as described below```
